### PR TITLE
Resource tests use helper functions

### LINKED ISSUE #77: Tests should use available helper functions
Old tests still create their tokens rather than use the base test's request_as_user() helper function. Using this will clean up the code.

### PR DESCRIPTION
**Pull request by PJvanVliet**
fixes #77 
- Cleaned up the code for resource tests with helper functions in base_test.py
- Didn't touch movements post tests, that would create a merge confict with #80 
- Extra: lets leader resource return 400 if no possible leaders are found for swap_leader.

### CODE REVIEW

File: web/gridt/tests/integration/resources/test_leader.py [hunk 1]
```diff
@@ -11,44 +11,60 @@
...
-            user1 = User("test1", "test1@test.com", "pass")
-            user2 = User("test2", "test2@test.com", "pass")
-            user3 = User("test3", "test3@test.com", "pass")
-            user4 = User("test4", "test4@test.com", "pass")
-            user5 = User("test5", "test5@test.com", "pass")
-
-            signal = Signal(user3, movement)
-            time_stamp = signal.time_stamp
+            movement = self.create_movement()
+            user1 = self.create_user_in_movement(movement)
```
Comment (marickmanrho) [inline web/gridt/tests/integration/resources/test_leader.py:17]: Maybe neater in a list comprehension or for loop.

File: web/gridt/tests/integration/resources/test_leader.py [hunk 2]
```diff
@@ -78,19 +91,14 @@ def test_swap_leader_movement_not_subscribed(self):
 
     def test_swap_leader_not_leader(self):
```
Comment (marickmanrho) [inline web/gridt/tests/integration/resources/test_leader.py:92]: I'm not sure what this is supposed to test. Can you explain it to me?

**Review by marickmanrho — CHANGES_REQUESTED** [review]
The tests should be made passing. Some things are a bit unclear to me, but in general it looks good!

File: web/gridt/tests/integration/resources/test_leader.py (hunk 2 above)
Comment (PJvanVliet) [reply web/gridt/tests/integration/resources/test_leader.py:92]: It is supposed to test that a user cannot request to swap a leader that they are not following.

File: web/gridt/tests/integration/resources/test_leader.py (hunk 1 above)
Comment (PJvanVliet) [reply web/gridt/tests/integration/resources/test_leader.py:17]: Good idea! The users don't actually need to be named either, so a for loop is good.

**Review by marickmanrho — APPROVED** [review]

File: web/gridt/tests/integration/resources/test_leader.py [hunk 3]
```diff
@@ -78,20 +79,11 @@ def test_swap_leader_movement_not_subscribed(self):
...
-
-            resp = self.client.post(
-                "/movements/Flossing/leader/2",
-                headers={"Authorization": f"JWT {token}"},
-            )
+            movement = self.create_movement()
+            user1 = self.create_user_in_movement(movement)
+            user2 = self.create_user()
+
+            resp = self.request_as_user(self.users[0], "POST", "/movements/1/leader/2")
```
Comment (Drvanon) [inline web/gridt/tests/integration/resources/test_leader.py:86]: Crazy how much that cleans up!

**Pull request closed without merging**